src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.dig;
+
+import foam.core.X;
+import foam.lib.json.JSONParser;
+import foam.lib.json.Outputter;
+import foam.lib.json.OutputterMode;
+import foam.nanos.dig.exception.*;
+import foam.nanos.http.HttpParameters;
+import foam.nanos.logger.Logger;
+import foam.util.SafetyUtil;
+import foam.nanos.http.WebAgent;
+
+import java.io.*;
+import java.nio.CharBuffer;
+import java.lang.Exception;
+import java.lang.reflect.*;
+import javax.servlet.http.HttpServletResponse;
+
+public class SugarWebAgent
+  implements WebAgent
+{
+  public SugarWebAgent() {}
+
+  public void execute(X x) {
```

Comment:
Please divide the execute function into smaller functions with descriptive names (something like steps), this makes it more readable/ enjoyable for the second dev.